sdk/src/main/java/io/radar/sdk/RadarVerificationManager.kt

```diff
@@ -1,59 +1,68 @@
 package io.radar.sdk
 
 import android.content.Context
-import com.google.android.play.core.integrity.IntegrityManagerFactory
-import com.google.android.play.core.integrity.IntegrityTokenRequest
+import android.location.Location
+import android.os.Build
+import android.util.Log
+import com.google.android.play.core.integrity.StandardIntegrityManager
+import com.google.android.play.core.integrity.StandardIntegrityManager.StandardIntegrityToken
+import com.google.android.play.core.integrity.StandardIntegrityManager.StandardIntegrityTokenRequest
+import com.google.android.gms.tasks.Task;
+import io.radar.sdk.RadarUtils.hashSHA256
 
 internal class RadarVerificationManager(
     private val context: Context,
     private val logger: RadarLogger,
 ) {
 
-    fun getIntegrityToken(googleCloudProjectNumber: Long?, nonce: String?, block: (integrityToken: String?, integrityException: String?) -> Unit) {
-        logger.d("Requesting integrity token")
-
-        if (googleCloudProjectNumber == null) {
-            val integrityException = "Missing Google Cloud project number"
-
-            logger.d(integrityException)
-
-            block(null, integrityException)
-
-            return
+    fun getRequestHash(location: Location, nonce: String): String {
+        val stringBuffer = StringBuilder()
+        // build a string of installId, latitude, longitude, mocked, nonce, sharing
+        stringBuffer.append(RadarSettings.getInstallId(this.context))
+        stringBuffer.append(location.latitude)
+        stringBuffer.append(location.longitude)
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+            //Note(travis): throw an error if SDK isn't JELLY_BEAN_MR2?
```

Comment:
@tderouin We should address and remove this note before we release (keeping in mind the SDK is open source and bad actors may look at this code)